packages/app-project/src/components/AuthModal/components/RegisterForm/RegisterForm.js

```diff
@@ -0,0 +1,235 @@
+import counterpart from 'counterpart'
+import React from 'react'
+import { Formik } from 'formik'
+import { Anchor, Box, Button, CheckBox, Heading, Text, FormField, TextInput } from 'grommet'
+import { SpacedText } from '@zooniverse/react-components'
+import Link from 'next/link'
+
+import en from './locales/en'
+
+counterpart.registerTranslations('en', en)
+
+const userNameFieldId = 'RegisterForm_username'
+const passwordFieldId = 'RegisterForm_password'
+const passwordConfirmFieldId = 'RegisterForm_password_confirm'
+const emailFieldId = 'RegisterForm_email'
+const realNameFieldId = 'RegisterForm_real_name'
+const privacyAgreementFieldId = 'RegisterForm_privacy_agreement'
+const emailListSignUpFieldId = 'RegisterForm_email_list_sign_up'
+const betaListSignUpFieldId = 'RegisterForm_beta_list_sign_up'
+const underageWithParentFieldId = 'RegisterForm_underage_with_parent'
+
+const Label = (props) => (
+  <SpacedText
+    weight='bold'
+    {...props}
+  />
+)
+
+const PrivacyPolicyLink = () => (
+  <Link href='/privacy' passHref>
+    <Anchor size='small'>{counterpart('RegisterForm.privacyLink')}</Anchor>
+  </Link>
+)
+
+function RegisterForm (props) {
+  const { onSubmit } = props
+  const initialValues = {
+    betaListSignUp: false,
+    email: '',
+    emailListSignUp: false,
+    password: '',
+    passwordConfirm: '',
+    privacyAgreement: false,
+    realName: '',
+    username: '',
+    underageWithParent: false
+  }
+  return (
+    <Box>
+      <Heading size='small' margin={{ bottom: 'xsmall', top: 'none' }}>
+        {counterpart('RegisterForm.heading')}
+      </Heading>
+      <Text>
+        {counterpart('RegisterForm.instruction')}
+      </Text>
+
+      <Formik
+        initialValues={initialValues}
+        onSubmit={onSubmit}
```

Comment:
we need a validation function to check:

- passwords are identical
- username is not taken?
- email hasn't already been used?

Formik can do async validation: https://jaredpalmer.com/formik/docs/guides/validation#validate